great_expectations/dataset/pandas_dataset.py

```diff
@@ -301,27 +301,31 @@ def expect_column_values_to_not_be_null(self, column,
         series = self[column]
         boolean_mapped_null_values = series.isnull()
 
-        element_count = int(len(series))
-        nonnull_values = series[boolean_mapped_null_values==False]
-        nonnull_count = int((boolean_mapped_null_values==False).sum())
+        element_count = len(series)
+        success_count = sum(~boolean_mapped_null_values)
+        unexpected_count = element_count - success_count
 
-        boolean_mapped_success_values = boolean_mapped_null_values==False
-        success_count = boolean_mapped_success_values.sum()
-
-        unexpected_list = [None for i in list(series[(boolean_mapped_success_values==False)])]
-        unexpected_index_list = list(series[(boolean_mapped_success_values==False)].index)
-        unexpected_count = len(unexpected_list)
+        unexpected_list = [None for i in range(unexpected_count)]
```

Comment:
I think there was another problem in the original and this should be: `[x for x in series[~boolean_mapped_null_values]]`, since it's possible that `np.nan` or another null value is there instead of `None`

Reply:
Makes sense. I didn't think about how there's a 100 ways to represent nothing

Reply:
I've been looking at `test_expect_column_values_to_not_be_null` and it was expecting for the `unexpected_list` to only have `None` values, regardless of the actual null value type. Furthermore, it seems that assertEqual can't assert equality of two `np.nan` values.

Reply:
This seems to be a bigger problem, I might just draft up a helper method to assert equality for arrays with multiple null value types. Numpy has some helper assertions, but they only like nan values - Nones are converted into NaN.

Reply:
OK. Let's disentangle these two things, and proceed with the remaining corrections to these expectations, but still just return a list of `None`s so we can close out the PR. (As I reflect--they'll currently all be converted by `recursively_convert_to_json_serializable` anyway I believe).